dimod/generators/mimo.py

```diff
@@ -790,46 +795,163 @@ def spin_encoded_comp(lattice: Union[int, nx.Graph],
                       seed: Union[None, int, np.random.RandomState] = None,
                       F_distribution: Union[None, str] = None,
                       use_offset: bool = False) -> dimod.BinaryQuadraticModel:
-    """Defines a simple coooperative multi-point decoding problem CoMP.
+    """Generate a coooperative multi-point (CoMP) decoding problem.
+
+    In `coordinated multipoint (CoMP) <https://en.wikipedia.org/wiki/Cooperative_MIMO>`_
+    neighboring cellular base stations coordinate transmissions and jointly 
+    process received signals.
+
+    Users each transmit complex valued symbols over a random channel :math:`F` of
+    some num_receivers, subject to additive white Gaussian noise. Given the received
+    signal y the log likelihood of a given symbol set :math:`v` is given by
+    :math:`MLE = argmin || y - F v ||_2`. When v is encoded as a linear
+    sum of spins the optimization problem is defined by a Binary Quadratic Model.
+
     Args:
-       lattice: A graph defining the set of nearest neighbor basestations. Each
-           basestation has ``num_receivers`` receivers and ``num_transmitters``
-           local transmitters. Transmitters from neighboring basestations are also
-           received. The channel F should be set to None, it is not dependent on the
-           geometric information for now.
-           Node attributes 'num_receivers' and 'num_transmitters' override the
-           input defaults.
-           lattice can also be set to an integer value, in which case a honeycomb
-           lattice of the given linear scale (number of basestations O(L^2)) is
-           created using ``_make_honeycomb()``.
-       modulation: modulation
-       integer_labeling:
-           When True, the geometric, quadrature and modulation-scale information
-           associated to every spin is compressed to a non-redundant integer label sequence.
-           When False, spin variables are labeled (in general, but not yet implemented):
-           (geometric_position, index at geometric position, quadrature, bit-precision)
-           In specific, for BPSK with at most one transmitter per site, there is 1
-           spin per lattice node with a transmitter, inherits lattice label)
-       F: Channel
-       y: Signal
-
-       See for ``spin_encoded_mimo`` for interpretation of other per-basestation parameters.
+        lattice: Geometry, as a :class:`networkx.Graph` or integer, defining 
+            the set of nearest-neighbor basestations. 
```

Comment:
should be nearest-neighbor locations (since we now allow basestations and/or receivers independently).